File: web_version/pmugg_files/guidelines/angledEdge.js
ms.angledEdge = function(position, angle, t) {
	this.position = position;
	this.angle = angle;
	// T is the coordinate along the direction of the edge tiling.
	this.t = t;
};

ms.angledEdge.RAY_LENGTH = 1000;

ms.angledEdge.prototype.copy = function() {
	return new ms.angledEdge(this.position.copy(), this.angle, this.t);
};

ms.angledEdge.prototype.getPosition = function() {
	return this.position;
};

ms.angledEdge.prototype.setPosition = function(position) {
	this.position = position;
};

ms.angledEdge.prototype.set = function(crossing) {
	this.position = crossing.getPosition();
	this.angle = crossing.getAngle();
};

ms.angledEdge.prototype.setAngle = function(angle) {
	this.angle = angle;
};

ms.angledEdge.prototype.getAngle = function() {
	return this.angle;
};

ms.angledEdge.prototype.setT = function(t) {
	this.t = t;
};

ms.angledEdge.prototype.getT = function() {
	return this.t;
};

ms.angledEdge.prototype.print = function() {
	return this.position.print();
};

ms.angledEdge.prototype.move = function(motion) {
	this.position.add(motion);
};

File: cpp_version/pmugg/guidelines/angled_edge.cpp
#include "angled_edge.h"

namespace ms {

AngledEdge::AngledEdge(const Vec2& position, float angle, float t)
    : position(position)
    , angle(angle)
    , t(t) {
}

AngledEdge* AngledEdge::copy() const {
    return new AngledEdge(position, angle, t);
}

Vec2 AngledEdge::getPosition() const {
    return position;
}

void AngledEdge::setPosition(const Vec2& newPosition) {
    position = newPosition;
}

float AngledEdge::getAngle() const {
    return angle;
}

void AngledEdge::setAngle(float newAngle) {
    angle = newAngle;
}

float AngledEdge::getT() const {
    return t;
}

void AngledEdge::setT(float newT) {
    t = newT;
}

void AngledEdge::set(const AngledEdge& crossing) {
    position = crossing.getPosition();
    angle = crossing.getAngle();
}

void AngledEdge::move(const Vec2& motion) {
    position += motion;
}

std::string AngledEdge::print() const {
    return position.print();
}

} // namespace ms 